### PR TITLE
Efficient map2world

### PR DESCRIPTION
**Pull request by manan-peppermint**
The updated code do the following task:

- It apply blurring technique like gaussian blur and median blur to remove noise.
- It detects contours and display the detected contour image for parameter tuning.
- It display preview of 3d world.
- Generate necessary world file, model file and mesh files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ map2gazebo Project URL: https://github.com/shilohc/map2gazebo
     - trimesh
     - matplotlib
     - pycollada
+    - shapely
```
Comment (ulkesh-peppermint) [inline README.md:14]: need to install mapbox-earcut, pyglet==1.5.29

File: src/map2world/map2world/map2world.py [hunk 2]
```diff
@@ -3,132 +3,404 @@
...
+            OccupancyGrid, map_topic, self.map_callback, qos_profile=map_sub_qos
+        )
+        # self.export_dir = str(os.getcwd())
         # Probably there's some way to get trimesh logs to point to ROS
         # logs, but I don't know it.  Uncomment the below if something
         # goes wrong with trimesh to get the logs to print to stdout.
-        #trimesh.util.attach_to_log()
+        # trimesh.util.attach_to_log()
+        
+        self.template_path = "~/Documents/map2world/src/map2world/templates/"
```
Comment (ulkesh-peppermint) [inline src/map2world/map2world/map2world.py]: Why is this path hardcoded?

File: src/map2world/map2world/map2world.py [hunk 3]
```diff
@@ -3,132 +3,404 @@
...
+                ) as f:
+                    mesh_line.export(f, "stl")
+                
             self.get_logger().info("Exported STL. Shutting down this node now.")
-        elif mesh_type == "dae":
-            with open(export_dir + "/map.dae", 'wb') as f:
-                f.write(trimesh.exchange.dae.export_collada(mesh))
+            
+            
+        if mesh_type == "dae":
```
Comment (ulkesh-peppermint) [inline src/map2world/map2world/map2world.py]: why is this changed from elif to if?

**Review by ulkesh-peppermint — CHANGES_REQUESTED** [review]

File: src/map2world/map2world/map2world.py (hunk 2 above)
Comment (manan-peppermint) [reply src/map2world/map2world/map2world.py]: Because relative path is not working. If generalised path is needed then i need to keep templates folder in share directory of map2world.

**Review by ulkesh-peppermint — COMMENTED** [review]
looks good
